training-react/tic-tac-toe/src/app/components/RouteCheck/components/ValidateRoute/index.js

```diff
@@ -0,0 +1,47 @@
+import React, { PureComponent } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import LoginService from '@services/loginService';
+import ROUTES from '@consts/route';
+
+class ValidateRoute extends PureComponent {
+  renderRoute = props => {
+    const { isPrivate, component: Component, isLogedIn } = this.props;
+    if (isLogedIn && !isPrivate) {
+      LoginService.setToken(localStorage.getItem('userToken'));
```

Comment:
U r setting the token in each render. That's wrong. U need set the token just once, when the user logged in. Move this set token from here.